### PR TITLE
Remove race conditions and simplify endpoint service constructor

### PR DESCRIPTION
**Pull request by arner**
This PR removes a few race conditions and simplifies the constructor for the endpoint service.

### CODE REVIEW

File: platform/view/core/endpoint/endpoint.go [hunk 1]
```diff
@@ -15,7 +15,6 @@ import (
 
 	"golang.org/x/exp/slices"
```
Comment (adecaro) [inline platform/view/core/endpoint/endpoint.go:17]: please, remove the blank line

File: platform/view/core/endpoint/endpoint.go [hunk 2]
```diff
@@ -241,20 +237,20 @@ func (r *Service) SetPublicKeyIDSynthesizer(publicKeyIDSynthesizer driver.Public
 }
 
 func (r *Service) pkiResolve(resolver *Resolver) []byte {
-	r.pkiExtractorsLock.RLock()
+	resolver.PKILock.Lock()
```
Comment (adecaro) [inline platform/view/core/endpoint/endpoint.go:240]: can we have a read lock here and if len(resolver.PKI) == 0 then we get a write lock?

**Review by adecaro — CHANGES_REQUESTED** [review]

File: platform/view/core/endpoint/endpoint.go [hunk 3]
```diff
@@ -241,20 +236,22 @@ func (r *Service) SetPublicKeyIDSynthesizer(publicKeyIDSynthesizer driver.Public
 }
 
 func (r *Service) pkiResolve(resolver *Resolver) []byte {
-	r.pkiExtractorsLock.RLock()
+	resolver.PKILock.RLock()
 	if len(resolver.PKI) != 0 {
-		r.pkiExtractorsLock.RUnlock()
```
Comment (adecaro) [inline platform/view/core/endpoint/endpoint.go:246]: missing read unlock

File: platform/view/core/endpoint/endpoint.go [hunk 4]
```diff
@@ -241,20 +236,22 @@ func (r *Service) SetPublicKeyIDSynthesizer(publicKeyIDSynthesizer driver.Public
...
 		return resolver.PKI
 	}
-	r.pkiExtractorsLock.RUnlock()
+	resolver.PKILock.RUnlock()
 
+	resolver.PKILock.Lock()
+	defer resolver.PKILock.Unlock()
 	resolver.PKI = r.ExtractPKI(resolver.Id)
```
Comment (adecaro) [inline platform/view/core/endpoint/endpoint.go:247]: check again if resolver.PKI is different from nil. Another thread might have already set it.

**Review by adecaro — CHANGES_REQUESTED** [review]

File: platform/view/services/comm/host/rest/websocket/stream_test.go [hunk 5]
```diff
@@ -104,19 +105,24 @@ func TestReader(t *testing.T) {
 	for _, message := range input {
 		assert.NoError(t, conn.ReadValue(message))
 	}
+	wg := sync.WaitGroup{}
```
Comment (adecaro) [inline platform/view/services/comm/host/rest/websocket/stream_test.go:108]: why the changes in this file? It seems unrelated to the rest.

**Review by adecaro — APPROVED** [review]
LGTM modulo a clarification.

File: platform/view/services/comm/host/rest/websocket/stream_test.go (hunk 5 above)
Comment (arner) [reply platform/view/services/comm/host/rest/websocket/stream_test.go:108]: This also caused the tests to fail when run with `-race`.
